packages/block-editor/src/components/inserter/use-async-list.js

```diff
@@ -4,9 +4,39 @@
 import { useEffect, useReducer } from '@wordpress/element';
 import { createQueue } from '@wordpress/priority-queue';
 
+/**
+ * Returns the first items from list that are present on state.
+ *
+ * @param {Array} list  New array.
+ * @param {Array} state Current state.
+ * @return {Array} First items present iin state.
+ */
+function getFirstItemsPresentInState( list, state ) {
+	const firstItems = [];
+	let index = 0;
+	let alreadyRendered = false;
+	do {
+		const item = list[ index ];
+		alreadyRendered = state.indexOf( item ) !== -1;
```

Comment:
Could use `Array#includes` if we're interested only in a boolean, not the index:

```suggestion
		alreadyRendered = state.includes( item );
```